flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java

```diff
@@ -167,12 +167,6 @@ public class HighAvailabilityOptions {
                     .withDescription(
                             "Defines the number of connection retries before the client gives up.");
 
-    @Documentation.Section(Documentation.Sections.EXPERT_ZOOKEEPER_HIGH_AVAILABILITY)
```

Comment:
Isn't the right process to deprecate the config option and remove it as part of 2.0? ...since configs are considered public API. :thinking:

Reply:
I agree that this Isn't the right process to deprecate the config option. What's special about it is that usually when we deprecate a config, the config can still take effect, but in this case, the config has no effect. So we might better just delete it to avoid user confusion, WDYT?

Reply:
I just found a similar case: the option high-availability.job.delay is deprecated and the comment states clearly that it has no effect on Flink any longer. That seems like the right process to do it. I'll follow the approach and update the PR. Thank you for pointing that out!

Reply:
Yeah, I see your point. The worst thing that could happen as far as I can think is that the field is used in some downstream project. I guess that would be a reason why we would follow the proper deprecation process. The only documentation I found about that is what's stated in the kind of old [Best Practices and Lessons Learned](https://cwiki.apache.org/confluence/display/FLINK/Best+Practices+and+Lessons+Learned)